packages/react-components/react-swatch-picker-preview/src/components/ColorSwatch/useColorSwatch.ts

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import { getIntrinsicElementProps, slot, useMergedRefs, useEventCallback } from '@fluentui/react-utilities';
+import type { ColorSwatchProps, ColorSwatchState } from './ColorSwatch.types';
+import { SwatchPickerOnSelectionChangeEvent } from '../SwatchPicker/SwatchPicker.types';
+import { useColorSwatchState_unstable } from './useColorSwatchState';
+import { useFocusWithin } from '@fluentui/react-tabster';
+import { useSwatchPickerContextValue_unstable } from '../../contexts/swatchPicker';
+
+/**
+ * Create the state required to render ColorSwatch.
+ *
+ * The returned state can be modified with hooks such as useColorSwatchStyles_unstable,
+ * before being passed to renderColorSwatch_unstable.
+ *
+ * @param props - props from this instance of ColorSwatch
+ * @param ref - reference to root HTMLDivElement of ColorSwatch
+ */
+export const useColorSwatch_unstable = (
+  props: ColorSwatchProps,
+  ref: React.Ref<HTMLButtonElement>,
+): ColorSwatchState => {
+  const { color, value } = props;
+
+  const context = useSwatchPickerContextValue_unstable();
+  const requestSelectionChange = context.requestSelectionChange;
+  const selected = context.selectedValue === value;
+  const onClick = useEventCallback((event: SwatchPickerOnSelectionChangeEvent) =>
+    requestSelectionChange({ event, selectedValue: value }),
+  );
+
+  const _role = 'radio';
+
+  const state: ColorSwatchState = {
+    components: {
+      root: 'button',
+    },
+    root: slot.always(
+      getIntrinsicElementProps('button', {
+        ref,
+        ...props,
+        role: props.role ?? _role,
```

Comment:
```suggestion
        role: props.role ?? 'radio',
```

Let's simplify for now